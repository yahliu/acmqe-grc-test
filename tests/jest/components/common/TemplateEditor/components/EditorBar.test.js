/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
'use strict'

import React from 'react'
import EditorBar from '../../../../../../src-web/components/common/TemplateEditor/components/EditorBar'
import renderer from 'react-test-renderer'

describe('EditorBar component', () => {
  it('renders as expected', () => {
    const fn = jest.fn()
    const exceptions = [{text:'bad', row:0}]
    const component = renderer.create(
      <EditorBar
        hasUndo={false}
        hasRedo={true}
        exceptions={exceptions}
        gotoEditorLine={fn}
        handleEditorCommand={fn}
        handleSearchChange={fn}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
