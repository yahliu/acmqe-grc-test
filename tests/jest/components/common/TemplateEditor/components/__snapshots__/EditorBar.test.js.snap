// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorBar component renders as expected 1`] = `
<div
  className="editor-bar"
>
  <div
    className="editor-bar-title"
  >
    Policy YAML
  </div>
  <div
    className="editor-bar-toolbar"
  >
    <div
      className="editor-bar-section"
    >
      <div
        aria-label="Reset"
        className="editor-bar-button "
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
        title="Reset"
      >
        <div>
          Reset
        </div>
      </div>
    </div>
    <div
      className="editor-bar-section"
    >
      <div
        aria-label="Undo"
        className="editor-bar-button  disabled"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
        title="Undo"
      >
        <svg>
          <use
            href="#diagramIcons_undo"
          />
        </svg>
      </div>
      <div
        aria-label="Redo"
        className="editor-bar-button "
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
        title="Redo"
      >
        <svg>
          <use
            href="#diagramIcons_redo"
          />
        </svg>
      </div>
    </div>
    <div
      className="editor-bar-section"
    >
      <div
        aria-label="Find"
        className="editor-bar-search"
        id="Find"
        role="region"
      >
        <div
          aria-labelledby="template-editor-search-undefined-label"
          className="bx--search bx--search--sm"
          role="search"
        >
          <svg
            alt=""
            aria-label=""
            className="bx--search-magnifier"
            fillRule="evenodd"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
          >
            <title>
              
            </title>
            <path
              d="M6.5 12a5.5 5.5 0 1 0 0-11 5.5 5.5 0 0 0 0 11zm4.936-1.27l4.563 4.557-.707.708-4.563-4.558a6.5 6.5 0 1 1 .707-.707z"
            />
          </svg>
          <label
            className="bx--label"
            htmlFor="template-editor-search-undefined"
            id="template-editor-search-undefined-label"
          >
            
          </label>
          <input
            aria-label="Find"
            className="bx--search-input"
            id="template-editor-search-undefined"
            onChange={[Function]}
            placeholder="Find"
            spellCheck={false}
            type="text"
          />
          <button
            className="bx--search-close bx--search-close--hidden"
            onClick={[Function]}
            type="button"
          >
            <svg
              alt="Provide a description that will be used as the title"
              aria-label="Provide a description that will be used as the title"
              fillRule="evenodd"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
            >
              <title>
                Provide a description that will be used as the title
              </title>
              <path
                d="M8 6.586L5.879 4.464 4.464 5.88 6.586 8l-2.122 2.121 1.415 1.415L8 9.414l2.121 2.122 1.415-1.415L9.414 8l2.122-2.121-1.415-1.415L8 6.586zM8 16A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"
              />
            </svg>
          </button>
        </div>
        <div
          aria-label="Previous"
          className="editor-bar-button  disabled"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
          title="Previous"
        >
          <svg>
            <use
              href="#diagramIcons_previous"
            />
          </svg>
        </div>
        <div
          aria-label="Next"
          className="editor-bar-button  disabled"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
          title="Next"
        >
          <svg>
            <use
              href="#diagramIcons_next"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="editor-bar-section"
    >
      <div
        aria-label="Close"
        className="editor-bar-button "
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
        title="Close"
      >
        <svg>
          <use
            href="#diagramIcons_close"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
